### PR TITLE
trilinos: switch to github url; fix missing py-numpy dependency

### PR DESCRIPTION
**Pull request by davydden**
fixes https://github.com/LLNL/spack/issues/1577
fixes https://github.com/LLNL/spack/issues/1578

12.6.4. (`trilinos+boost^boost+python+mpi`) tested on:

- [x] macOS 10.11 + clang + gfortran-6.1
- [x] Ubuntu 14.04 + gcc-6.1

### CODE REVIEW

File: var/spack/repos/builtin/packages/trilinos/package.py [hunk 1]
```diff
@@ -43,18 +43,22 @@ class Trilinos(Package):
...
+    version('12.4.2', '4c25a757d86bde3531090bd900a2cea8')
+    version('12.2.1', '85d011f7f99a776a9c6c2625e8cb721c')
+    version('12.0.1', 'bcb3fdefd14d05dd6aa65ba4c5b9aa0e')
+    version('11.14.3', 'dea62e57ebe51a886bee0b10a2176969')
+    version('11.14.2', 'e7c3cdbbfe3279a8a68838b873ad6d51')
+    version('11.14.1', 'b7760b142eef66c79ed13de7c9560f81')
+
+    def url_for_version(self, version):
+        return '%s/trilinos-release-%s.tar.gz' % \
+            (Trilinos.base_url, str(version).replace('.', '-'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/trilinos/package.py]: `str(version).replace('.', '-')` is equivalent to `version.dashed()`

Comment (davydden) [inline var/spack/repos/builtin/packages/trilinos/package.py]: did not know that, will change...

**Comment by davydden** [comment]
@Rombur @KineticTheory @goxberry : if you have your favourite Trilinos version in Spack, feel free to test the updated urls :smile: . The archives are not the same, but hopefully everything works.

File: var/spack/repos/builtin/packages/trilinos/package.py (hunk 1 above)
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/trilinos/package.py]: Yep, there's also one for dotted (the default) and underscored. It's all in `lib/spack/spack/version.py` if you're interested.

Comment (davydden) [inline var/spack/repos/builtin/packages/trilinos/package.py]: hm, i got an error that `str` object is not callable with `version.dashed()`. Weird...

Comment (davydden) [inline var/spack/repos/builtin/packages/trilinos/package.py]: i have a feeling that `url_for_version` takes in string object, not the `Version` object.

Comment (davydden) [inline var/spack/repos/builtin/packages/trilinos/package.py]: however on my mac it looks like this is indeed `Version`. If i try to do `version.replace()` then i have
```
AttributeError: 'Version' object has no attribute 'replace'
```

Comment (adamjstewart) [inline var/spack/repos/builtin/packages/trilinos/package.py]: Hmm, that's weird. The version parameter should always be of type Version...

Comment (adamjstewart) [inline var/spack/repos/builtin/packages/trilinos/package.py]: Ah, figured it out, sorry. It's `version.dashed`, not `version.dashed()`. It's a property, not a method.

Comment (davydden) [inline var/spack/repos/builtin/packages/trilinos/package.py]: @adamjstewart thanks!

**Comment by adamjstewart** [comment]
LGTM

**Comment by davydden** [comment]
@tgamblin the Trilinos PR is ready to be merged. Previously used url is still down, thus i think we should switch to Github ASAP.

**Pull request merged**